Realm/Realm/Configurations/FlexibleSyncConfiguration.cs

```diff
@@ -182,20 +182,22 @@ internal override Native.SyncConfiguration CreateNativeSyncConfiguration()
         /// </summary>
         /// <value>The <see cref="ClientResetHandlerBase"/> that will be used to handle a client reset.</value>
         /// <remarks>
-        /// Currently, Flexible sync only supports the <see cref="ManualRecoveryHandler"/>. Support for <see cref="DiscardLocalResetHandler"/> will come in the future.
+        /// Currently, Flexible sync only supports the <see cref="ManualRecoveryHandler"/>. Support for the newer modes will come in the future.
         /// </remarks>
         /// <exception cref="NotSupportedException">
-        /// Flexible sync is still in beta, so at the moment <see cref="DiscardLocalResetHandler"/> is not supported.
+        /// Flexible sync is still in beta, so only <see cref="ManualRecoveryHandler"/> is supported.
         /// </exception>
         /// <seealso href="https://docs.mongodb.com/realm/sdk/dotnet/advanced-guides/client-reset/">Client reset docs</seealso>
         public override ClientResetHandlerBase ClientResetHandler
         {
             get => base.ClientResetHandler;
             set
             {
-                if (value is DiscardLocalResetHandler)
+                if (value is DiscardUnsyncedChangesHandler ||
+                    value is RecoverUnsyncedChangesHandler ||
+                    value is RecoverOrDiscardUnsyncedChangesHandler)
```

Comment:
It's kind of a moot point considering we're going to support all reset handler types soon, but this is not a great way to check. Apart from checking by type, it's also missing a check for `DiscardLocalResetHandler`.